.travis.yml

```diff
@@ -161,7 +161,11 @@ jobs:
 
       script:
         - |
-          # Run integration tests.
+          version="$(./linkerd version --client --short | tr -cd '[:alnum:]-')"
+          # Cleanup previous integration test artifacts if present
+          ./bin/test-cleanup linkerd
```

Comment:
I think we should do the same as in `after_script`:
```
./bin/test-cleanup linkerd-$version
```
as the intention would be to get rid of garbage from previous test integration runs that didn't finish and thus couldn't run `after_script`

Reply:
I think that should take care of that since `./bin/test-cleanup` will delete all namespaces with the `linkerd-*` prefix This includes `linkerd-$version`. So that should clean up namespaces that have been left behind by other tests.

Reply:
I agree with @alpeb that the integration test should only delete what it creates. AFAIK, the integration test doesn't create the `linkerd` namespace, right? If one exists, it's probably a dev namespace; i.e. others might run the integration test alongside their WIP dev stuff.

Reply:
Ah! I think I get it now. @ihcsim @alpeb  you are right in that this would interfere with other linkerd installs that are running on the same cluster.  Thanks for the explanation @ihcsim!